robot/client/client.go

```diff
@@ -935,3 +942,33 @@ func (rc *RobotClient) StopAll(ctx context.Context, extra map[resource.Name]map[
 	_, err := rc.client.StopAll(ctx, &pb.StopAllRequest{Extra: e})
 	return err
 }
+
+// ModuleLog sends a log entry to the server. To be used by Golang modules wanting to log over gRPC and not
+// by normal Golang SDK clients.
+func (rc *RobotClient) ModuleLog(ctx context.Context, log zapcore.Entry, fields []zapcore.Field) error {
+	// NOTE(benjirewis): formatting string is my best guess at what would be the
+	// useful pieces of the whole time stamp for module users (month, day, hour,
+	// minute, second).
+	timeStr := log.Time.Format("Jan 02 15:04:05")
+	message := fmt.Sprintf(`%v %q {%q: %v`, log.Caller.TrimmedPath(), log.Message, moduleTSLogKey, timeStr)
+	for _, field := range fields {
+		message = fmt.Sprintf("%v, %v: %v", message, field.Key, field.String)
+	}
+	message = fmt.Sprintf("%v}", message) // close }
```

Comment:
This field formatting and insertion of the module log timestamp is hacky... I couldn't find a great way around this, and I found it easier to stringify things on the client side rather than handle everything on the server side after converting to and from proto.

Reply:
Cool; filed RSDK-6280 and referenced it here.